Add changes detection feature to cd visibility docs (#28108)
DataDog/documentation: content/en/continuous_delivery/features/changes_detection.md

---

> title: Changes Detection

Reviewer: (will need to be changed in multiple places)

I think `Changes Detection` is too hard to interpret. I recommend to align it with our UI: `Code Changes`.

To detect the changes deployed, Datadog runs the [`git log`][1] between the current deployment commit SHA and the previous deployment commit SHA. Merge commits are excluded from the computation.

The deployed changes are visible inside any deployment execution of the [Deployment Executions page][2]. The **Code Changes** tab shows the previous deployment taken into consideration, and the changes detected between the two.

{{< img src="continuous_delivery/features/code_changes_tab.png" alt="Code Changes tab for changes detection feature" style="width:100%;">}}

Additionally, the **Deployments** column of the [Recent Code Changes][3] page displays the service and environment details for all deployments that included a specific commit. This view provides a quick way to understand if and where your changes are deployed.
Hovering over the service value reveals whether the deployment has reached all expected environments, based on where the service is typically deployed.

{{< img src="continuous_delivery/features/recent_code_changes_deployments.png" alt="Showing deployments in Recent Code Changes page" style="width:100%;">}}

Changes are only detected for deployments that:

> - Have a service and an environment. Additionally, the service needs to have the file path specs defined in Software Catalog (see the [setup instructions](#specify-service-file-path-patterns) for more information).
> - Have a repository URL and a commit SHA defined.

Reviewer: ```suggestion
- Have a service and an environment.
- Have a repository URL and a commit SHA defined.
- The service has file path specs defined in Software Catalog (see the [setup instructions](#specify-service-file-path-patterns) for more information).
```

Reviewer: Maybe you can switch 1 and 2 so that the 2 service requirements are close to the other
